Sources/Pill/Pill.swift

```diff
@@ -0,0 +1,110 @@
+import SwiftUI
+
+extension Warp {
+    /// Used to visually label objects on a page for quick recognition. They allow users to filter a list or label information. 
+    public struct Pill: View, Hashable {
+        public static func == (lhs: Warp.Pill, rhs: Warp.Pill) -> Bool {
+            lhs.text == rhs.text &&
+            lhs.isClosable == rhs.isClosable &&
+            lhs.style == rhs.style
+        }
+        
+        public func hash(into hasher: inout Hasher) {
+            hasher.combine(text)
+            hasher.combine(isClosable)
+            hasher.combine(style)
+        }
+        
+        /// Pill text.
+        private let text: String
+        /// Triggered when Pill is tapped.
+        private let onTap: () -> Void
+        /// Triggered when Pill is closed.
+        private let onClose: (() -> Void)?
+        /// The content description of the close icon. Used for accessibility purposes.
+        private let iconContentDescription: String?
+        /// Pill style.
+        private let style: Warp.PillStyle
+        /// Object that will provide needed colors.
+        private let colorProvider: ColorProvider = Warp.Config.colorProvider
+
+        public init(
+            text: String,
+            onTap: @escaping () -> Void = {},
+            onClose: (() -> Void)? = nil,
+            iconContentDescription: String? = nil,
+            style: Warp.PillStyle = .filter
+        ) {
+            self.text = text
+            self.onTap = onTap
+            self.onClose = onClose
+            self.iconContentDescription = iconContentDescription
+            self.style = style
+        }
+
+        public var body: some View {
+            SwiftUI.Button {
+                onTap()
+            } label: {
+                HStack(spacing: 8) {
+                    Text(text)
+                    if isClosable {
+                        SwiftUI.Button {
+                            onClose?()
+                        } label: {
+                            Image(systemName: "xmark")
+                        }
```

Comment:
When a pill is closable,
Shouldn't we handle the updates?
Or is it the callee's responsibility?!

- If it's our responsibility:
We have to add a boolean state property, when close button is tapped, we will update that boolean property and return nothing as this view's body, in that way pill is completely removed from the view hierarchy.

Reply:
Probably the caller, we still don't know how the users will use these, so probably will do that later on when we have more concrete information how our users will use this.